test/common/upstream/upstream_impl_test.cc

```diff
@@ -1429,6 +1429,30 @@ TEST_F(StaticClusterImplTest, MalformedHostIP) {
       "setting cluster type to 'STRICT_DNS' or 'LOGICAL_DNS'");
 }
 
+TEST_F(StaticClusterImplTest, NoHostsTest) {
+
+  const std::string yaml = R"EOF(
+    name: staticcluster
+    connect_timeout: 0.25s
+    load_assignment:
+      cluster_name: foo
+      endpoints:
+      - priority: 1
+  )EOF";
+
+  envoy::api::v2::Cluster cluster_config = parseClusterFromV2Yaml(yaml);
+  Envoy::Stats::ScopePtr scope = stats_.createScope(fmt::format(
+      "cluster.{}.", cluster_config.alt_stat_name().empty() ? cluster_config.name()
```

Comment:
I think this can be simplified because we know the above config and whether it has an alt stat name (there is none).

Reply:
Done. Testing for the alt stat name was just the pattern in the test file.